dev/attest/src/assertions/assertions.ts

```diff
@@ -1,7 +1,7 @@
 import * as assert from "node:assert/strict"
 import { isDeepStrictEqual } from "node:util"
 
-import { snapshot, stringify } from "../../../../src/utils/serialize.js"
+import { snapshot, stringify } from "../../../utils/src/serialize.js"
```

Comment:
Did it not seem viable to limit external imports to `main`? I'd love if the build failed for other imports like this, but we can always add it later if it was an issue.

Reply:
I have this working, I'm going to open this change as a separate PR because A) it might lead to other discussions, and B) it nearly doubles the number of changed files

Reply:
Just in case, here's the config I used to get that working.

Note that I had to comment out everything in "paths" because it looks like that _might_ not have been working before -- probably because of some interference between `no-restricted-imports` and `@typescript-eslint/no-restricted-imports`

```json
        "@typescript-eslint/no-restricted-imports": [
            "error",
            {
                paths: [
                    // "error",
                    // "node:assert",
                    // "node:buffer",
                    // "node:child_process",
                    // "node:cluster",
                    // "node:crypto",
                    // "node:dgram",
                    // "node:dns",
                    // "node:domain",
                    // "node:events",
                    // "node:freelist",
                    // "node:fs",
                    // "node:http",
                    // "node:https",
                    // "node:module",
                    // "node:net",
                    // "node:os",
                    // "node:path",
                    // "node:node:punycode",
                    // "node:querystring",
                    // "node:readline",
                    // "node:repl",
                    // "node:smalloc",
                    // "node:stream",
                    // "node:string_decoder",
                    // "node:sys",
                    // "node:timers",
                    // "node:tls",
                    // "node:tracing",
                    // "node:tty",
                    // "node:url",
                    // "node:util",
                    // "node:vm",
                    // "node:zlib"
                ],
                patterns: [
                    {
                        group: ["../../dev/utils/src/*", "!../../dev/utils/src/main.js"],
                        message: "Deep importing from \"dev/utils/*\" is not allowed, please import directly from \"dev/utils/src/main.js\" instead"
                    },
                    {
                        group: ["../../dev/attest/*", "!../../dev/attest/src/main.js"],
                        message: "Deep importing from \"dev/attest/*\" is not allowed, please import from \"dev/attest/src/main.js\" instead"
                    }
                ],
           }
        ],
```